app/src/androidTest/java/com/conupods/Settings/SettingsActivityTest.java

```diff
@@ -0,0 +1,86 @@
+package com.conupods.Settings;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.view.View;
+import android.widget.CheckBox;
+
+import androidx.test.rule.ActivityTestRule;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import com.conupods.OutdoorMaps.View.MapsActivity;
+import com.conupods.R;
+
+import static androidx.test.core.app.ApplicationProvider.getApplicationContext;
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.intent.Intents.intended;
+import static androidx.test.espresso.intent.matcher.IntentMatchers.hasComponent;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static org.junit.Assert.*;
+
+public class SettingsActivityTest {
```

Comment:
I'm not sure if you copied this test from online but can you make sure that the tests do not depend on each other? For example, if you were to perform the tests in a different order, would they still pass? If the answer is no, because you rely on a succession of button clicks in a certain order, then you only need 1 test for that order of buttons.